e2e/nomostest/registryproviders/image.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registryproviders
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"kpt.dev/configsync/e2e/nomostest/gitproviders"
+	"kpt.dev/configsync/e2e/nomostest/testshell"
+)
+
+// use an auto-incrementing index to create unique file names for tarballs
+var imageIndex int
+
+// NewImage creates a new OCIImage object and associated tarball using the provided
+// Repository. The contents of the git repository will be bundled into a tarball
+// at the artifactDir. The resulting OCIImage object can be pushed to a remote
+// registry using its Push method.
+func NewImage(artifactDir string, shell *testshell.TestShell, repository *gitproviders.Repository, provider RegistryProvider) (*OCIImage, error) {
```

Comment:
`BuildImage` maybe? It's not just a Go constructor, so using New- is a little confusing.

Reply:
Done